Observer/Order/Save.php

```diff
@@ -148,6 +152,24 @@ public function execute(Observer $observer)
                 if ($nostoCustomer instanceof NostoCustomer) {
                     $nostoCustomerId = $nostoCustomer->getNostoId();
                 }
+                // If the id is still null, fetch the `customer_reference`
+                if ($nostoCustomerId === null &&
```

Comment:
I'd at least wrap this new block into a private function. The execute method is already too long.